stripe/_http_client.py

```diff
@@ -1209,8 +1210,13 @@ def close(self):
 class HTTPXClient(HTTPClient):
     name = "httpx"
 
+    _client: Optional["HTTPXClientType"]
+
     def __init__(
-        self, timeout: Optional[Union[float, "HTTPXTimeout"]] = 80, **kwargs
+        self,
+        timeout: Optional[Union[float, "HTTPXTimeout"]] = 80,
+        allow_sync_methods=True,
```

Comment:
Should this be the default? It seems like unintentionally introducing synchronous methods (especially via a pagination request) is easier to do, so maybe we should guard against that?

Reply:
Hmm, that does make sense! It's also less of a trapdoor, if people report getting confused/inconvenienced by sync methods being disabled, it's non-breaking to change this default from False to True but not the other way around.

Reply:
(Will change)

Reply:
Are we sure? Sync is a fully supported feature for HTTPX. Maybe customers want to use https for other reasons than async methods.

Reply:
That's a fair point. We never got a feature request for HTTPX outside of async support context, though, and I see "httpx-but-not-async" as a fairly niche use case. 

I am willing to accept requiring users who `stripe.default_http_client = stripe.HTTPXClient()`and **do** intend to use sync methods to apply an easy-to-discover remedy, as a fair price for preventing users who `stripe.default_http_client = stripe.HTTPXClient()` and **don't** intend to use sync methods from making a hard-to-discover mistake. Especially since this decision is the easier to reserve.